### DISCUSSION TITLE
Unbundling external dependencies

### DISCUSSION THREAD

**Discussion by selfisekai** [ghostty-org/ghostty#3171] [Q&A] [answered]
The `pkg` directory ships build scripts for multiple third party dependencies. There should be an option available to use distro-provided versions, in line with distro requirements such as this: https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies

**Comment by mitchellh** [comment, marked as answer]
We do: https://github.com/ghostty-org/ghostty/blob/main/PACKAGING.md

Unless I'm misunderstanding. The build scripts are only used for local builds for ease of development. But we've documented how to use full dynamic linking specifically for packaging as you noted.

**Reply by selfisekai** [reply]
Ok, I do see now that the `-fsys=` options are there, but that's not mentioned in the file you linked

**Reply by mitchellh** [reply]
The `--system` flag forces this for all packages to ensure an offline build (which is mentioned).
